pyproject.toml

```diff
@@ -29,3 +61,59 @@ ignore_missing_imports = true
 disallow_untyped_calls = true
 disallow_untyped_defs = true
 disallow_incomplete_defs = true
+
+[tool.poetry.dependencies]
+python = ">=3.7,<3.11"
```

Comment:
I intend to comment more about the various discussions around, but having a quick moment, I wonder here about this poetry-dependent part, when there are `[project] dependencies` and  `[project.optional-dependencies]` which are build-tool agnostic.  Does poetry not support the more general options now?

Reply:
I don't know.
I basically copied out of pymedphys.
I will take a more careful look at the poetry documentation (it seems a bit skimpy on the content of the pyproject.toml).

I am going to split out the sphinx fix, and also the deprecated github workflow fix.
That will effectively leave this PR for just the pyproject.toml changes